apps/prairielearn/src/pages/instructorStudents/components/StudentsCard.tsx

```diff
@@ -0,0 +1,145 @@
+
+const columnHelper = createColumnHelper<StudentRow>();
+
+// This default must be declared outside the component to ensure referential
+// stability across renders, as `[] !== []` in JavaScript.
+const DEFAULT_SORT: SortingState = [];
+
+export function StudentsCard({ students }: { students: StudentRow[] }) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+  const [columnVisibility, setColumnVisibility] = useState<ColumnVisibilityState>({});
+  const [columnPinning, setColumnPinning] = useState<ColumnPinningState>({
+    left: [],
+    right: [],
+  });
+
+  const columns = useMemo<ColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('created_at', {
+        header: 'Enrolled At',
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
+        },
+      }),
+    ],
+    [],
+  );
+
+  const table = useReactTable({
+    data: students,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.uid,
+    state: {
+      sorting: sorting ?? undefined,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+    },
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      minSize: 150,
+      size: 300,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  return (
+    <div class="card d-flex flex-column h-100">
+      <div class="card-header bg-primary text-white">
+        <div class="d-flex align-items-center justify-content-between">
+          <div>Students</div>
+          <div>
+            <DownloadButton students={students} table={table} />
+          </div>
+        </div>
+      </div>
+      <div class="card-body d-flex flex-column">
+        <div class="d-flex flex-row mb-2">
```

Comment:
Nit, for the filter and view config, I'd be inclined to copy the behavior of GitHub's issue page:

- Input spans full width.
- Button wraps to second line when the viewport is too narrow to accommodate both.

Reply:
![CleanShot 2025-06-25 at 11 48 30@2x](https://github.com/user-attachments/assets/8dadf36a-6a62-42d3-bcd6-306292f6e8f6)
![CleanShot 2025-06-25 at 11 48 51@2x](https://github.com/user-attachments/assets/e85a17b0-11e2-4bc5-8226-5a2232760598)

Changed in 8202aa92aa7b5f0b69e45da2eb2c4f5210984383.